src/Common/src/Interop/Unix/libgss/Interop.GssApi.cs

```diff
@@ -0,0 +1,162 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Diagnostics;
+using System.Runtime.InteropServices;
+using System.Text;
+using Microsoft.Win32.SafeHandles;
+
+internal static partial class Interop
+{
+    internal static class GssApi
+    {
+        internal static bool EstablishSecurityContext(
```

Comment:
Same comment as before - the managed code which calls into it will be part of a separate PR. The flow would be along the lines of https://github.com/vijaykota/corefx/commit/18f24701de1cd26909630bdcadfe16ce27f2259a